src/gpio.rs

```diff
@@ -518,6 +538,51 @@ macro_rules! gpio {
                         Ok(unsafe { (*$GPIOX::ptr()).idr.read().bits() & (1 << $i) == 0 })
                     }
                 }
+
+                impl<MODE> ExtiPin for $PXi<Input<MODE>> {
+                    /// Configure EXTI Line $i to trigger from this pin.
+                    fn make_interrupt_source(&mut self, afio: &mut AFIO) {
+                        let offset = 4 * ($i % 4);
+                        afio.$exticri.modify(|r, w| unsafe {
+                            let mut exticr = r.bits();
+                            exticr = (exticr & !(0xf << offset)) | ($extigpionr << offset);
+                            w.bits(exticr)
+                        });
+                    }
+
+                    /// Generate interrupt on rising edge, falling edge or both
+                    fn trigger_on_edge(&mut self, exti: &mut EXTI, edge: Edge) {
+                        match edge {
+                            Edge::RISING => {
+                                exti.rtsr.modify(|r, w| unsafe { w.bits(r.bits() | (1 << $i)) });
```

Comment:
There already exist access methods for all the registers. For instance, example for EXTI1 w/o macros woud look like this: 
```
        exti.imr.modify(|_, w| w.mr1().set_bit());
        exti.ftsr.modify(|_, w| w.tr1().set_bit());
        exti.rtsr.modify(|_, w| w.tr1().set_bit());
```
Is it possible to use the same access methods in order to avoid the use of ```unsafe``` block ?

Reply:
That's a good point, I was thinking about that myself.
However, I don't think using the mr1/mr2/mr... registers is feasible, since that would make it very difficult to generate such code in the first place. Currently, the macro for the pins just gets a pin number and with that, it is able to set the proper bits for imr, ftsr and rtsr (and other stuff outside of my patch).
Doing anything else would make the macro invocations much more lengthy and, more importantly, error-prone. It would be easy to slip in a copy-paste-mistake where pin3 uses mr3, but tr2 by accident.

Also, since this is inside of a library, I don't see it as a problem here to use `unsafe`. All in all I'd say it's not worth the hassle.

Reply:
Sounds reasonable to me

Reply:
Sounds reasonable to me as well. Thanks for clarification!